### PR TITLE
hw6 task1 - instances_counter decorator - code and tests

### PR DESCRIPTION
**Pull request by IuliiaLanchuk**

### CODE REVIEW

File: homework6/task1_counter/tests/test_instance_counter.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+@instances_counter
+class User:
+    pass
+
+
+@instances_counter
+class Automobile:
+    color = "black"
```
Comment (Cjkjvfnby) [inline homework6/task1_counter/tests/test_instance_counter.py]: Lets rename `color` to the `instances_amount`

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: homework6/task1_counter/tests/test_instance_counter.py (hunk 1 above)
Comment (IuliiaLanchuk) [reply homework6/task1_counter/tests/test_instance_counter.py]: A, ok, you probably mean instances_amount in ClassHelper should be private - __instances_amount. Fixed

Comment (Cjkjvfnby) [reply homework6/task1_counter/tests/test_instance_counter.py]: This feature was added to Python exactly for that case.

File: homework6/task1_counter/tests/test_instance_counter.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def test_user_class_create_instances_and_reset_instance():
+    assert User.get_created_instances() == 0
+    user, _, _ = User(), User(), User()
+    assert user.get_created_instances() == 3
+    assert user.reset_instances_counter() == 3
+
+
+def test_user_create_instances_reset_and_then_check_instances_amount():
```
Comment (Cjkjvfnby) [inline homework6/task1_counter/tests/test_instance_counter.py]: It is better to describe not what are you doing (I can guess it from code) but hat are you want to check. 

```
test_reset_all_set_instance_to_zero
```

File: homework6/task1_counter/tests/test_instance_counter.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    user, _ = User(), User()
+    assert user.reset_instances_counter() == 2
+    assert user.get_created_instances() == 0
+
+
+def test_automobile_class_create_and_reset_instance():
```
Comment (Cjkjvfnby) [inline homework6/task1_counter/tests/test_instance_counter.py]: What is the difference between `automobile_class` and `user_class`?  What additional aspects of the code this test covers?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: homework6/task1_counter/tests/test_instance_counter.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    assert Automobile.get_created_instances() == 0
+    automobile = Automobile(2005)
+    assert automobile.get_created_instances() == 1
+    assert automobile.reset_instances_counter() == 1
+
+
+def test_automobile_class_attributes_and_methods_do_not_change():
+    opel_astra = Automobile(1998)
+    assert opel_astra.instances_amount == "black"
```
Comment (Cjkjvfnby) [inline homework6/task1_counter/tests/test_instance_counter.py:35]: Anything special with this attribute? Why do you test it?

File: homework6/task1_counter/tests/test_instance_counter.py (hunk 3 above)
Comment (IuliiaLanchuk) [reply homework6/task1_counter/tests/test_instance_counter.py]: Agree, it is the same one

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**